scipy/stats/_stats_py.py

```diff
@@ -3956,7 +3957,109 @@ def __init__(self, msg=None):
         self.args = (msg,)
 
 
-def pearsonr(x, y):
+def _pearsonr_fisher_ci(r, n, confidence_level, alternative):
+    """
+    Compute the confidence interval for Pearson's R.
+
+    Fisher's transformation is used to compute the confidence interval
+    (https://en.wikipedia.org/wiki/Fisher_transformation).
+    """
+    if r == 1:
+        zr = np.inf
+    elif r == -1:
+        zr = -np.inf
+    else:
+        zr = np.arctanh(r)
+    se = np.sqrt(1 / (n - 3))
+
+    if alternative == "two-sided":
+        h = special.ndtri(0.5 + confidence_level/2)
+        zlo = zr - h*se
+        zhi = zr + h*se
+        rlo = np.tanh(zlo)
+        rhi = np.tanh(zhi)
+    elif alternative == "less":
+        h = special.ndtri(confidence_level)
+        zhi = zr + h*se
+        rhi = np.tanh(zhi)
+        rlo = -1.0
+    else:
+        # alternative == "greater":
+        h = special.ndtri(confidence_level)
+        zlo = zr - h*se
+        rlo = np.tanh(zlo)
+        rhi = 1.0
+
+    return ConfidenceInterval(low=rlo, high=rhi)
+
+
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+
+PearsonRResultBase = _make_tuple_bunch('PearsonRResultBase',
+                                       ['statistic', 'pvalue'], [])
+
+
+class PearsonRResult(PearsonRResultBase):
+    """
+    Result of `scipy.stats.pearsonr`
+
+    Attributes
+    ----------
+    statistic : float
+        Pearson product-moment correlation coefficent.
+    pvalue : float
+        P-value.
+    alternative : str
+        The value of `alternative` that was passed to `pearsonr`.
+    n : int
+        The length of `x` and `y` that were given to `pearsonr`.
+
+    Methods
+    -------
+    confidence_interval(confidence_level=0.95)
```

Comment:
@mdhaber

> I want to bed able to control which methods of this class get rendered. Right now, it seems to be all or none.

```suggestion
    confidence_interval
```

`numpydoc` can't parse `confidence_interval(confidence_level=0.95)` which is why the documentation of the `confidence_interval` method isn't linked (but it is generated). `numpydoc` only expects the method name and not the signature so changing it this way will link its docs.

Reply:
Thanks!